sedaro/src/sedaro/results/simulation_result.py

```diff
@@ -47,15 +47,15 @@ def templated_agents(self) -> List[str]:
         return tuple([
             entry['name'] for _, entry
             in self.__meta['structure']['scenario']['blocks'].items()
-            if entry['type'] == 'Agent' and not entry['peripheral']
+            if entry['type'] == 'TemplatedAgent'
```

Comment:
Nit but future us will thank current us if we update this to check "is `TemplatedAgent` or a subclass of `TemplatedAgent`". 

|@zduvall what do you think about adding a `block_dicts_by_type` utility to the python client and then using something like this to add another utility `is_block_instance(block, type)` which behaves like pythons `isinstance` but for our blocks instead.

Reply:
We could use this check on line 58 (`if not is_block_instance(entry, 'TemplatedAgent')`)

The existence of a `block_dicts_by_type` util would allow us to generalize `_agent_in_type_supers` too.

Reply:
Sure, happy to do something like that. Sounds like a good idea. Would you want me to work on that for this branch? Or after it's merged?

Reply:
I've made a fix to use that internal function to check for `TemplatedAgent` and `PeripheralAgent` too

Reply:
Thanks guys!  @rtumin2 your changes are great.  This is ready to me merged!